noop.go

```diff
@@ -68,7 +70,25 @@ func (n NoopTracer) StartSpanWithOptions(opts StartSpanOptions) Span {
 	return defaultNoopSpan
 }
 
+// Extractor belongs to the Tracer interface.
+func (n NoopTracer) Extractor(format interface{}) Extractor {
+	return defaultNoopInjectorExtractor
+}
+
+// Injector belongs to the Tracer interface.
+func (n NoopTracer) Injector(format interface{}) Injector {
+	return defaultNoopInjectorExtractor
+}
+
 // JoinTrace belongs to the Tracer interface.
 func (n NoopTracer) JoinTrace(operationName string, parent interface{}) Span {
 	return defaultNoopSpan
 }
+
+func (n noopInjectorExtractor) InjectSpan(span Span, carrier interface{}) error {
+	return nil
+}
+
+func (n noopInjectorExtractor) JoinTrace(operationName string, carrier interface{}) (Span, error) {
```

Comment:
I like the idea of defining a

```
var TraceNotFound = errors.New("trace not found by Extractor")
```

... so your example above would still work, but a user may be able to tell the difference between the not-found case and the parse-error case...